src/__tests__/reporter/testIemReporting.spec.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ *  Copyright 2021 EPAM Systems
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+import MyReporter from '../../reporter';
+import { mockConfig } from '../mocks/configMock';
+import { mockedResult, mockedTestParams, RPClientMock } from '../mocks/RPClientMock';
+import { FinishTestItemObjType, StartTestObjType } from '../../models';
+import { TEST_ITEM_TYPES } from '../../constants';
+import * as utils from '../../utils';
+
+describe('start test reporting', () => {
+  jest.spyOn(utils, 'getConfig').mockImplementation(() => mockConfig);
+  const reporter = new MyReporter();
+  reporter.client = new RPClientMock(mockConfig);
+  reporter.launchId = 'tempLaunchId';
+  const parentId = 'tempTestItemId';
+  const startTestObj: StartTestObjType = {
+    startTime: reporter.client.helpers.now(),
+    name: mockedTestParams.title,
+    type: TEST_ITEM_TYPES.STEP,
+  };
+  const mockedTestItems = new Map([['test', { id: 'tempTestItemId', name: 'test' }]]);
+  const spyOnTestBegin = jest.spyOn(reporter, 'onTestBegin');
```

Comment:
Are you sure you need to track of the call to this function? You call it right in the line below and you know exactly how many times it has been called.

Reply:
When i `expect(reporter.onTestBegin)` i get `Matcher error: received value must be a mock or spy function`